### ISSUE TITLE
Upgrade to latest release

### ISSUE THREAD

**Issue by NobodyXu** [oeb25/binswap#4]
In the latest release, we've refactored binstalk again and extract several crates out:
 - `binstalk-registry` for fetching crates from crates.io or any custom registry
 - `binstalk-fetchers` for actually fetching crates from release
 - `binstalk-bins` for installing bins
 - `simple-git` for cloning git repository directly
 - `cargo-toml-workspace` for loading manifest of any crate inside a workspace

With that, you could remove `binstalk` and disable features that are unused and further customize the APIs.

I also think it's good idea for such crates to be maintained in cargo-binstall repository since we would keep it up to date.

**Comment by oeb25** [comment]
These changes sound great!

> I also think it's good idea for **such crates** to be maintained in cargo-binstall repository since we would keep it up to date.

Which crates are you referring to here? If you are referring to moving binswap to cargo-binstall I would be happy to help in the process!

**Comment by NobodyXu** [comment]
> Which crates are you referring to here? If you are referring to moving binswap to cargo-binstall I would be happy to help in the process!

Yes, I'm referring to `binswap`, I personally think it makes sense for it to be maintained in `cargo-binstall`.

Also, I think cargo-bins need more maintainers for these crates and I welcome any PR and anybody who want to be one.

@passcod What's your thought on this?

**Comment by passcod** [comment]
happy to

**Comment by NobodyXu** [comment]
@oeb25 Can you submit a PR to `cargo-binstall`?
It will take quite some time since we would have to review it.
You could split it into multiple PRs.

**Comment by NobodyXu** [comment]
I was a bit busy and almost forgot about this 😂

**Comment by NobodyXu** [comment]
cc @oeb25 Are you still intere in upstreaming this into cargo-binstall repository?

I can give you maintainer role if you want.

**Comment by NobodyXu** [comment]
cc @oeb25 friendly ping on this issue, I really believe this should be maintained in cargo-binstall repository to make sure it gets all the updates of binstalk.
